frontend/appflowy_flutter/lib/plugins/database/widgets/cell/editable_cell_skeleton/url.dart

```diff
@@ -214,29 +214,19 @@ class MobileURLEditor extends StatelessWidget {
 }
 
 void openUrlCellLink(String content) async {
-  String url = "";
+  late Uri uri;
 
   try {
-    // check protocol is provided
-    const linkPrefix = [
-      'http://',
-      'https://',
-      'file://',
-      'ftp://',
-      'ftps://',
-      'mailto:',
-    ];
-    final shouldAddScheme =
-        !linkPrefix.any((pattern) => content.startsWith(pattern));
-    url = shouldAddScheme ? 'http://$content' : content;
-
-    // get hostname and check validity
-    final uri = Uri.parse(url);
-    final hostName = uri.host;
-    await InternetAddress.lookup(hostName);
+    uri = Uri.parse(content);
+    if (!uri.hasScheme || uri.scheme == "localhost") {
+      uri = Uri.parse('http://$content');
+      await InternetAddress.lookup(uri.host);
+    }
   } catch (_) {
-    url = "https://www.google.com/search?q=${Uri.encodeComponent(content)}";
+    uri = Uri.parse(
+      "https://www.google.com/search?q=${Uri.encodeComponent(content)}",
+    );
   } finally {
-    await afLaunchUrlString(url);
+    await launchUrl(uri);
```

Comment:
```suggestion
    await afLaunchUrl(uri);
```

Reply:
@Xazin @LucasXu0 While I was implementing this logic, I didn't like the behavior with `afLaunchUrl`'s `addingHttpSchemeWhenFailed` in certain cases. If I don't provide a scheme and I call it after parsing it, like below

```
final uri = Uri.parse('instagram.com');
await afLaunchUrl(uri, addingHttpSchemeWhenFailed: true);
```

`afLaunchUrl` tries to immediately launch it. When that fails or returns false, it then adds the http scheme and retry, resulting in this even though on the second try the link launches successfully.

![image](https://github.com/AppFlowy-IO/AppFlowy/assets/71320345/cc028ce4-9b03-4bac-ab74-dfc1b9c99097)

So here in this function, I did the handling myself and removed the need for `afLaunchUrl`. I also needed to support searching with google if the link cannot open.

How should I proceed? Should I make any changes to `afLaunchUrl?

Reply:
> I also needed to support searching with google if the link cannot open.

This logic should not be included in `afLaunchUrl` function. I have reviewed your changes. It's ok to keep your logic.

> afLaunchUrl tries to immediately launch it. When that fails or returns false, it then adds the http scheme and retry, resulting in this even though on the second try the link launches successfully.

I suggest skipping the first return if addingHttpSchemeWhenFailed is set to true.